core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorConcurrencyTest.scala

```diff
@@ -112,6 +113,13 @@ class GroupCoordinatorConcurrencyTest extends AbstractCoordinatorConcurrencyTest
     }.toSet
   }
 
+  /**
+   * handleTxnCommitOffsets does not complete delayed requests now so it causes error if handleTxnCompletion is executed
+   * before completing delayed request. In random mode, we use this global lock to prevent such error.
+   */
+  private var isRandomOrder = false
```

Comment:
Instead of introducing a global var, could we add a new param when constructing CommitTxnOffsetsOperation and CompleteTxnOperation?